File: 0221.cpp
#include<iostream>
#include<string>
#include<list>
using namespace std;

int N, M;
list<int> players;

bool check(int n, string s)
{
  if(n % 5 == 0 && n % 3 == 0)
    return s == "FizzBuzz";
  if(n % 5 == 0)
    return s == "Buzz";
  if(n % 3 == 0)
    return s =="Fizz";
  return s == to_string(n);
}

int main()
{
loop:
  cin >> M >> N;
  if(!N)
    return 0;

  players.clear();
  for(int i = 0; i < M; i++)
    players.push_back(i + 1);
  auto p = players.begin();
  for(int i = 1; i <=N; i++)
  {
    string s;
    cin >> s;
    if(players.front() == players.back())
      continue;
    if(!check(i, s))
      p = players.erase(p);
    else 
      ++p; 
    if(p == players.end())
      p = players.begin();
  }
  for(auto v : players)
  {
    cout << v;
    if(v != players.back())
      cout << " ";
  }
  cout << endl;
  goto loop;
}

File: 0221.py
while 1:
  m, n = map(int, raw_input().split())
  if not m:
    break
  p = range(m)
  t = 0
  for i in range(1, n + 1):
    s = raw_input()
    if(i % 15 == 0 and s != "FizzBuzz" or
       i % 5 == 0 and s != "Buzz" or
       i % 3 == 0 and s != "Fizz" or
       i % 3 and i % 5 and s != str(i)):
      p.pop(t)
    else:
      t = (t + 1) % len(p)
  for i in range(len(p) - 1):
    print p[i] + 1,
  print p[len(p) - 1] + 1
